### ISSUE TITLE
Parameter extraction failed

### ISSUE THREAD

**Issue by ysshah95** [sparkfun/SparkFun_MLX90640_Arduino_Example#4]
Hello,
I have been using MLX90640 with Arduino Due and after uploading the Basic_Reading program I get a message in serial monitor saying "Parameter extraction failed" and the pixel values are displayed as nanC. 

Any ideas what I might try?

Thank You
Yash Shah

**Comment by nseidle** [comment]
Hmm. That is odd. I'm not entirely sure but let's eliminate a handful of things:

* What MLX90640 board are you using?
* How are you powering it?
* Do you have resistors on the I2C lines?

It sounds kind of like I2C is not working but I would have expected the code to fail [earlier](https://github.com/sparkfun/SparkFun_MLX90640_Arduino_Example/blob/master/Firmware/Example1_BasicReadings/Example1_BasicReadings.ino#L50).

**Comment by ysshah95** [comment]
- I am using Qwiic IR Array - MLX90640 110° FOV by SparkFun [https://www.sparkfun.com/products/14768]
- I am Powering Arduino Due with micro USB.
- Yeah, I have 2 pull up resistors (2.2k each) on I2C  lines.

**Comment by nseidle** [comment]
Ok - The Qwiic board already has pull up resistors so you shouldn't need additional ones (shouldn't hurt though).  Have you got a logic analyzer? Can you confirm the Due's I2C port is performing as expected?

**Comment by ysshah95** [comment]
I don't have the logic analyzer but I will somehow confirm the working of Due's I2C port. Do you think changing the controller to Teensy 3.5 would be the perfect option if I2C of Due is not working?

**Comment by ysshah95** [comment]
I took a help from this Arduino Forum (https://playground.arduino.cc/Main/I2cScanner) to check the working of I2C of Due, and to my surprise, the I2C of Due is working and it displays "I2C device found at address 0x33  !", message every 5 seconds.

**Comment by nseidle** [comment]
Good - that means the sensor is at least acknowledging I2C traffic.

Changing the controller to a Teensy would eliminate one unknown. I'll see if I can get a Due and test things out.

**Comment by ysshah95** [comment]
Okay, that would be a great help. Thank you so much.

**Comment by Eternia-Labs** [comment]
Same problem as above with an ESP32

**Comment by ysshah95** [comment]
Okay. So, is it recommended to change the sensor?

**Comment by Eternia-Labs** [comment]
No, the sensor is perfectly fine, so is the code here. I haven't tweaked it yet. Will get back with results for ESP32 Arduino .. Meanwhile this might help you with a raspberry pi [https://github.com/pimoroni/mlx90640-library](https://github.com/pimoroni/mlx90640-library)

**Comment by ghost** [comment]
The issue is definitively related to the Wire implementation in the Due. I had the same problem but switched to a Teensy 3.2. I didn't investigate the problem to the end but you can search the web for Due & I2C issues.

**Comment by nseidle** [comment]
Whew. I've spent some time on this and cannot reach a resolution. The Due I2C library and hardware seem buggy.

![image](https://user-images.githubusercontent.com/117102/43729399-67807ad2-9965-11e8-8c3a-f2a87b2a364d.png)

Above is an image of the Due glitching the clock line. Wire and Wire1 ports both exhibit this bad behavior. 

Attached is the code I was using to test. This is as far as I can take it at the moment.

**Comment by nseidle** [comment]
Hmm. Here's the code again.

[Example3_DueExample.zip](https://github.com/sparkfun/SparkFun_MLX90640_Arduino_Example/files/2268321/Example3_DueExample.zip)

**Comment by ysshah95** [comment]
Thank you so much for the help. I really appreciate your effort. I recently changed my controller from Due to Teensy 3.5, and the library works fine. 

But, I also tried the latest Due example you attached in the last comment, but still, it doesn't work. I think it gets stuck in one of the function "MLX90640_DumpEE". I have attached the screenshot of the terminal.

![capture](https://user-images.githubusercontent.com/31979840/43973367-5175af80-9ca5-11e8-9a7b-bb1c900fd044.PNG)
